### PR TITLE
38/hoc types

### LINKED ISSUE #31: Port to react-query 4.x.x to support React 18
It seems that react-query 3.x.x isn't compatible with React 18. Therefore, we should consider porting the apps-query-client to react-query 4.x.x in order to be able to port the apps to React 18.

### LINKED ISSUE #38: Reconcile types for new HOC

### PR DESCRIPTION
**Pull request by spaenleh**
This PR adds corresponding types to the HOC (`withToken` and `withContext`) such that they correspond to the hooks they use. 

### Changes:
- The context (`LocalContext`) changed from a `Map` to a `Record` (type is `RecordOf<LocalContext>`) through the use of the `LocalContextRecord` factory.
- `Context` enum is renamed to `CONTEXTS` in order to not conflict with the LocalContext's `Context` export in `index.ts`

### Fixes: 
- MirageJS dates are now in ISO format 

closes #38 
with 399a0f1 we can now close #31

### CODE REVIEW

File: src/mockServer/mockServer.ts [hunk 1]
```diff
@@ -48,6 +48,7 @@ export const mockServer = ({
     deleteAppDataShouldThrow?: boolean;
   };
 } = {}) => {
+  console.log('Mock server created, with appContext:', appContext);
```
Comment (pyphilia) [inline src/mockServer/mockServer.ts:51]: Remove console.

**Review by pyphilia — CHANGES_REQUESTED** [review]
Thanks for the types 🏗️  Let me know when you're done with what we discussed (use Contexts from graasp-utils)

**Comment by spaenleh** [comment]
This PR now uses React18 and react-query v4 in beta. And we now use `microbundle` instead of `microbundle-crl`.

**Review by pyphilia — APPROVED** [review]
👏

**Pull request merged**